apps/ledger-live-mobile/src/components/CustomImage/ImageProcessor.tsx

```diff
@@ -0,0 +1,164 @@
+import { Flex } from "@ledgerhq/native-ui";
+import React from "react";
+import { WebView } from "react-native-webview";
+import { WebViewErrorEvent } from "react-native-webview/lib/WebViewTypes";
+import { ImageProcessingError } from "./errors";
+import { injectedCode } from "./injectedCode/imageProcessing";
+import { InjectedCodeDebugger } from "./InjectedCodeDebugger";
+import { ImageBase64Data, ImageDimensions } from "./types";
+
+export type ProcessorPreviewResult = ImageBase64Data & ImageDimensions;
+
+export type ProcessorRawResult = { hexData: string } & ImageDimensions;
+
+export type Props = ImageBase64Data & {
+  onError: (error: Error) => void;
+  onPreviewResult: (arg: ProcessorPreviewResult) => void;
+  onRawResult: (res: ProcessorRawResult) => void;
+  /**
+   * number >= 0
+   *  - 0:  full black
+   *  - 1:  original contrast
+   *  - >1: more contrasted than the original
+   * */
+  contrast: number;
+  debug?: boolean;
+};
+
+/**
+ * Component to do some processing on an image (apply grayscale with 16 gray
+ * levels, apply some contrast, output a preview and a hex reprensentation of
+ * the result)
+ *
+ * It:
+ *  - takes as an input an image base64 data & a contrast value as an input
+ *  - displays nothing (except for a warning in __DEV__ if the injected code is
+ *  not correctly injected)
+ *  - outputs a preview of the result in base64 data uri
+ *  - on user confirmation (call `ref.requestRawResult()`) it outputs the raw
+ *  result which is a hex representation of the 16 levels of gray image.
+ *
+ *
+ * Under the hood, this is implemented with a webview and some code injected in
+ * it as it gives access to some web APIs (Canvas & Image) to do the processing.
+ *
+ * */
+export default class ImageProcessor extends React.Component<Props> {
+  /**
+   * NB: We are a class component here because we need to access some methods from
+   * the parent using a ref.
+   */
+
+  webViewRef: WebView<{}> | null = null;
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.contrast !== this.props.contrast) this.setAndApplyContrast();
+    if (prevProps.imageBase64DataUri !== this.props.imageBase64DataUri)
+      this.computeResult();
+  }
+
+  handleMessage = ({ nativeEvent: { data } }: any) => {
```

Comment:
i think the type is `WebViewMessageEvent` from [here](https://github.com/react-native-webview/react-native-webview/blob/37c8d34df602d8c693ad261ac99b87b5cd2d4b57/src/WebViewTypes.ts#L1203)

Reply:
And to follow the other "handle" functions, maybe rename it to `handleWebViewMessage` ? (i'm picky 🐩)

Reply:
ah yep thanks 👍 the "any" typing is probably a remain of the POC days haha